runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/Environments.java

```diff
@@ -112,6 +116,20 @@ public String specification() {
     }
 
     public static JavaVersion forSpecification(String specification) {
+      for (JavaVersion ver : JavaVersion.values()) {
+        if (ver.specification.equals(specification)) {
+          return ver;
+        }
+      }
+      JavaVersion fallback = java11;
```

Comment:
Instead of always falling back to 11, could we find the nearest LTS version to fall back to?

Reply:
Sure, working on this right now

Reply:
I can see 2 strategies:

1. find the absolute nearest. lower or higher
2. find the nearest higher. if you have code compile in java 13 you can hope to have it working on java 17 but certainly not on java 11

It's still possible to have compiled your code with 11 but run it on java 13 jvm. this is why solution 1 can work and I'm going with it for now.

Reply:
one last question is what to do for the middle point, for now I using the higher version (e.g.: use 17 when invoked from 14)